### DISCUSSION TITLE
How can I set `colorTheme` as part of a custom variant?

### DISCUSSION THREAD

**Discussion by dgrcode** [chakra-ui/chakra-ui#6407] [Q&A]
Reading about [responsive variant in the docs](https://chakra-ui.com/docs/styled-system/customize-theme#responsive-variants) I found this:
 > Based on how the responsive variants are designed, there's no support for responsive `colorScheme` since it is technically not a variant.

Which makes me think this might not be possible.

I was hoping to be able to do something like the following, but it didn't work:
```js
import { extendTheme } from '@chakra-ui/react'
const theme = extendTheme({
  colors: {
    brand: {
      // some colors
    }
  },
  components: {
    Button: {
      variants: {
        buy: {
          defaultProps: {
            colorTheme: 'brand'
          }
        }
      }
    }
  }
})
```

**Comment by esitarz** [comment]
@dgrcode did you ever get to the bottom of this?

[withDefaultColorScheme](https://chakra-ui.com/docs/styled-system/customize-theme#theme-extension-withdefaultcolorscheme) gets closer, but isn't quite there...

**Reply by dgrcode** [reply]
Nope, I didn't manage to find the solution I was looking for. From a quick look a the docs it seems nothing has changed around this concept since I posted this discussion, so maybe it's still unsupported.

My guess is that there might be a workaround, but I ended up applying the styles to the component manually.

**Comment by keithburgie** [comment]
You can do something like this:
```typescript
export default {
  variants: {
    buy: (props: StyleFunctionProps) =>
      props.theme.components.Button.variants.solid({
        ...props,
        colorScheme: 'brand',
      }),
  },
  defaultProps: {
    variant: 'buy',
  },
};
```

**Reply by dgrcode** [reply]
that looks interesting. I'll definitely try next time I hit this situation. Thanks for sharing! :)
